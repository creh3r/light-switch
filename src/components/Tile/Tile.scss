.tile {
  width: 50px;
  height: 50px;
  border: 2px solid map-get($theme-colors, greyDarkened);
  cursor: pointer;
  outline: none;

  &--isActive {
    background-color: map-get($theme-colors, activeLight);
  }
}