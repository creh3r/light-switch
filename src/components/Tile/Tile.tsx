import React from 'react';

type TileProps = {
  isActive: boolean,
  clickHandler: () => void,
};

/**
 * Render a grid tile
 *
 * @param isActive - Check if current tile is active
 * @param children - The click handler
 */
function Tile({ isActive, clickHandler }: TileProps) {
  return (
    <div
      onClick={() => clickHandler()}
      onKeyDown={() => clickHandler()}
      role="button"
      aria-label="Tile"
      tabIndex={0}
      className={(isActive ? 'tile tile--isActive' : 'tile')}
    />
  );
}

export default Tile;
