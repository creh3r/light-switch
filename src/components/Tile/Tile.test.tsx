// eslint-disable-next-line import/no-unresolved
import { shallow } from 'enzyme';
import React from 'react';
import Tile from './Tile';

describe('<Tile /> component', () => {
  it('... should match snapshot', () => {
    const wrapper = shallow(<Tile />);
    expect(wrapper).toMatchSnapshot();
  });

  it('... should have active class', () => {
    const wrapper = shallow(<Tile isActive />);
    expect(wrapper.find('.tile--isActive')).toHaveLength(1);
  });
});
