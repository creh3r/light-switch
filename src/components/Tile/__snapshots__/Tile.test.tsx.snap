// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> component ... should match snapshot 1`] = `
<div
  aria-label="Tile"
  className="tile"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
/>
`;
